examples/apps/mouse-drawing/src/main.rs

```diff
@@ -0,0 +1,99 @@
+use color_eyre::Result;
+/// A Ratatui example that demonstrates how to handle mouse events.
+///
+/// This example demonstrates how to handle mouse events in Ratatui. The example allows you to draw
+/// lines by clicking and dragging the mouse.
+///
+/// This example runs with the version of Ratatui found in the branch that you are currently
+/// reading. It may not work with the released version of Ratatui. See the `latest` branch for the
+/// latest code which works with the released version of Ratatui.
+use crossterm::{
+    event::{self, DisableMouseCapture, EnableMouseCapture, KeyCode, MouseEvent, MouseEventKind},
+    execute,
+};
+use ratatui::{
+    layout::{Position, Rect, Size},
+    symbols,
+    text::Line,
+    DefaultTerminal, Frame,
+};
+
+fn main() -> Result<()> {
+    color_eyre::install()?;
+    let terminal = ratatui::init();
+    let result = MouseDrawingApp::default().run(terminal);
+    ratatui::restore();
+    result
+}
+
+#[derive(Default)]
+struct MouseDrawingApp {
+    pub should_exit: bool,
```

Comment:
I tend to avoid adding unnecessary docs when things seem obvious, and here I can't think of any documentation that adds anything useful that's not already conveyed by the name.

Reply:
added